feat(system): add system packages

sfoster1 opened the pull request:
Add a whole bunch of packages:
- avahi
- dropbear
- nginx
- boot scripts
- the random libs for audio and video
- proper network handling

### Networking

Wired networks are set up to try for dhcp once and, on timeout, fall back to negotiating link-local addresses. This is done at the NetworkManager layer because even though avahi is also able to do this, it does weird stuff with aliased network interfaces that NetworkManager strongly dislikes. 

The way we do it in networkmanager is to remove the default ethernet handling, then use the autoconnect priority feature of network configs to have a dhcp config (wired) and a link-local config (wired-linklocal) in the correct priority order. These should never be modified and in fact will be restored on boot; if we add something that requires configuring wired interfaces, like static ip setups, we should add new connections at autoconnect priority 2.

### Systemd stuff

Added an opentrons.target and changed our default target to it. This lets us remember what to add wants to rather than just multi-user.target. Right now the only things that are wanted by opentrons.target is the boot.d scripts, but there will be more - for instance, this is where the unit file that starts the api server will be linked.
Related to https://github.com/Opentrons/opentrons/issues/2873

sfoster1 committed:
feat(system): Add various little system capabilities

- avrdude for module updates
- ffmpeg and mpg123 for camera and audio
- a cute motd

sfoster1 committed:
feat(system): Add dropbear

Dropbear serves ssh for us. It is configured to only allow logins with ssh keys.
Right now, a single ssh public key is put in authorized_keys during build. In
the future, we'll remove this for release builds and implement ways to add and
remove keys from local connections only.

sfoster1 committed:
feat(system): Add avahi and advertise on mdns

This implements our mdns advertising using static configurations. We'll
advertise as opentrons-(hostname).

sfoster1 committed:
feat(system): add nginx

Adds nginx and the nginx.conf from the monorepo (but without specifying the pid,
since that's handled by the default unit file).

sfoster1 committed:
feat(system): fall back to linklocal if dhcp fails on eth0

The easiest way to do this is to configure two connections in NetworkManager.
The first, with a higher priority, is for dhcp and gives up if it fails; the
second, with a normal priority, is for link-local and tries forever (until the
carrier goes away). This handles both kinds of connection.

sfoster1 committed:
feat(system): Run boot.d scripts on startup

Executable scripts in /data/boot.d will be run after filesystems are mounted.

sfoster1 committed:
feat(system): Add opentrons.target as default and fixup systemd

It always feels a little weird just using multi-user.target for things that
aren't really strictly necessary for multi-user work; and it was definitely
weird to be using graphical.target since we don't have graphics; so we have
opentrons.target as our new default systemd target, and added the thing that
runs boot.d as a want.

Also, fixup a couple other annoying systemd things we hadn't previously caught:
- systemd ExecStart parsing doesn't work like a shell unless you specifically
say to run it in a shell
- as much as you might want it to, putting a WantedBy in a unit only does
something if you're running systemctl enable on it, and we're not, so we need
som explicit symlinks
- it turns out echo and cat are different and we are now successfully setting up
authorized keys
- home is /root not /home

btmorr reviewed (approved):
Reviewed f2f, looks good

sfoster1 committed:
feat(system): Add smoothie programmer

This is available on github at https://github.com/capiman/lpc21isp but I don't want to add it as a package because the checked in binary already works.

mcous commented on board/opentrons/ot2/rootfs-overlay/etc/avahi/services/opentrons.service:
```diff
@@ -0,0 +1,15 @@
+
+<service-group>
+
+  <name replace-wildcards="yes">opentrons-%h</name>
```
(Assuming this is the mDNS name of the service) The `opentrons-...` prefix no longer serves a purpose. Historically, it was added before we were advertising our own mDNS service and relying on the fact the Resin would also advertise it's SSH service (so we couldn't filter by looking for our own port).

If we don't want to use the `opentrons-` prefix anymore, we don't have to

sfoster1 replied:
I want to still because _http._tcp isn't just our service (although the port is unique, but still, it's not like we're advertising _opentrons.tcp like we perhaps ought to be).

mcous commented on board/opentrons/ot2/rootfs-overlay/etc/nginx/nginx.conf:
```diff
@@ -0,0 +1,43 @@
+            proxy_set_header         Upgrade $http_upgrade;
+            proxy_set_header         Connection "upgrade";
+            proxy_read_timeout       1h;
+            proxy_pass               http://unix:/tmp/aiohttp.sock;
```
Are we planning on continuing the use of this UNIX socket?

sfoster1 replied:
For now. May change later.

mcous commented on board/opentrons/ot2/rootfs-overlay/var/home/.ssh/robot_key.pub:
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCmd34g9YqwQ7oMOJjPJezLIAOsov9vtr/dWqrwsVPLX1MkM0zFJg1Ou9NE9p30VaDnE92Hx0N9ndr8mznoImx/l8lkm9Z2QW+uO1eSKoQVwlz/1Dc+MJq8gy41GWYlKLr7p2b0TicAPcvGGYsFOmEUszOr5M78kOa2SHcHryd5Ol0T1bZQJ9p66JM8yY0upclrZouJEI5ozq/fHcihuR/WVo/dq81lM3lImfnfTe48exV42CGMndfTM0LFwRSRN1ANNr+Wto1a+X4lRnlk/RFzxB8a49UxNqQTCpdy/49wilQV7FJu9oHsaJI1dcKV9DE657gOrzM7tBaJXL6yh8Jn seth@Seths-MacBook ...
```
> seth@Seths-MacBook-Air.local

Was this intentional?

mcous reviewed (approved):
💠 a few questions but overall looks good. Lemme know if you need any more detail on the mDNS name thing

sfoster1 committed:
fixup: remove hostname comment from pubkey
